imap/append.c

```diff
@@ -865,6 +865,33 @@ static int append_apply_flags(struct appendstate *as,
     return r;
 }
 
+struct findstage_cb_rock {
+    const char *partition;
+    char *stagefile;
+    size_t size;
+};
+
+static int findstage_cb(const conv_guidrec_t *rec, void *vrock)
+{
+    struct findstage_cb_rock *rock = vrock;
+    mbentry_t *mbentry = NULL;
+
```

Comment:
This needs a

```if (rec->part) return 0;```

Otherwise we could wind up pointing to an email which has this GUID as an attachment by mistake.